### PR TITLE
[JENKINS-45315] TLS support - enables cloud hosted mongo

### PR DESCRIPTION
**Pull request by timja**

### CODE REVIEW

**Comment by timja** [comment]
cc @rsandell @TWestling

**Comment by timja** [comment]
Retriggering build for incrementals

**Comment by timja** [comment]
and again

**Comment by timja** [comment]
ping @rsandell @TWestling

File: docs/azure.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+Azure Cosmos DB requires an index when an order by query is performed, in order for this plugin to work you need to run the following commands:
+
+the example uses the mongo shell, but you can use any mongo client (except the one in the Azure portal which doesn't work for this)
+
+```bash
+mongo '<primary-connection-string'
+
+use <db-name> # this should be the db name you chose earlier, e.g. bfa
+db.failureCauses.createIndex({ "name": 1 })
```
Comment (rsandell) [inline docs/azure.md:68]: can't the plugin add this index somehow?

Comment (timja) [reply docs/azure.md:68]: most likely, didn't think of that.

**Pull request merged**